crux-bench/src/crux/bench/watdiv.clj

```diff
@@ -0,0 +1,483 @@
+(ns crux.bench.watdiv
+  (:require [clojure.java.io :as io]
+            [clojure.tools.logging :as log]
+            [crux.bench :as bench]
+            [crux.api :as crux]
+            [crux.index :as idx]
+            [crux.rdf :as rdf]
+            [crux.sparql :as sparql]
+            [datomic.api :as d]
+            [crux.io :as cio]
+            [clojure.string :as string])
+  (:import [java.io Closeable File StringReader]
+           java.util.Date
+           org.eclipse.rdf4j.query.Binding
+           org.eclipse.rdf4j.repository.RepositoryConnection
+           org.eclipse.rdf4j.repository.sail.SailRepository
+           org.eclipse.rdf4j.rio.RDFFormat
+           org.eclipse.rdf4j.sail.nativerdf.NativeStore))
+
+(defn output-to-file [out output]
```

Comment:
best not to open the file, write a line to it, close the file every time - we can use a writer instead (one to pair on if you're not familiar with the pattern)

```clojure
(with-open [w (io/writer (io/file ...))]
  (doseq [m maps]
    (.write w (prn-str m))))
```
general pattern is ^^, but in this case you won't have a `doseq` - just write for each output line

Reply:
Also, if there is still a file, the current method will just append result sets to the end - was wondering how id do this, will give using a writer a go :)